src/transformers/quantizers/quantizer_torchao.py

```diff
@@ -0,0 +1,157 @@
+            if "torch_dtype" in kwargs:
+                self.torch_dtype = kwargs["torch_dtype"]
+            else:
+                self.torch_dtype = torch.float32
+                logger.info("Setting torch_dtype to torch.float32 as the default value since it was not specified.")
+
+    def update_torch_dtype(self, torch_dtype):
+        if torch_dtype is None:
+            torch_dtype = torch.float32
+        return torch_dtype
+
+    def check_quantized_param(
+        self,
+        model: "PreTrainedModel",
+        param_value: "torch.Tensor",
+        param_name: str,
+        state_dict: Dict[str, Any],
+        **kwargs,
+    ) -> bool:
+        module, tensor_name = get_module_from_name(model, param_name)
+
+        return isinstance(module, torch.nn.Linear) and (tensor_name == "weight")
+
+    def create_quantized_param(
+        self,
+        model: "PreTrainedModel",
+        param_value: "torch.Tensor",
+        param_name: str,
+        target_device: "torch.device",
+        state_dict: Dict[str, Any],
+        unexpected_keys: List[str],
+    ):
+        """
+        Each nn.Linear layer is processsed here.
+        We first check if the corresponding module state_dict contains already torchao quantized parameters.
+        If not, we create a temp linear layer with the module state_dict params and use it for quantization
+        """
+        module, tensor_name = get_module_from_name(model, param_name)
+
+        layer_name = param_name.replace(".weight", "").replace(".bias", "")
+        parent_module = find_parent(model, layer_name)
+        node = layer_name.split(".")[-1]
+
+        # Step 0: set module state_dict
+        module_state_dict = {key.split(".")[-1]: state_dict[key] for key in state_dict if layer_name in key}
+
+        # Step 1: populate module with weight/bias from module state dict
+        for key in module_state_dict:
+            setattr(module, key, torch.nn.Parameter(module_state_dict[key]))
+
+        # Step 2: Replace module with either HQQLinear or move it to device. We do this via setattr on the parent as doing on it on the module
+        # directly doesn't work.
+
+        _STR_TO_METHOD = {
+            "int4_weight_only": int4_weight_only,
+            "int8_weight_only": int8_weight_only,
+            "int8_dynamic_activation_int8_weight": int8_dynamic_activation_int8_weight,
+        }
+
+
+        module = module.to(dtype=self.torch_dtype, device=target_device)
+        setattr(parent_module, node, module)
+
+        if self.quantization_config is not None:
+            assert self.quantization_config.quant_type in _STR_TO_METHOD.keys(), f"Requested quantization type: {self.quantization_config.quant_type} is not supported yet, please add support in TorchAOHfQuantizer."
+            quantize_(module, _STR_TO_METHOD[self.quantization_config.quant_type](**self.quantization_config.kwargs))
+            setattr(parent_module, node, module)
+
+        torch.cuda.empty_cache()
```

Comment:
why do we need this?

Reply:
not sure, this is copied from hqq quantizer, I can remove